model/src/application/interaction/mod.rs

```diff
@@ -74,6 +86,33 @@ impl Interaction {
             Self::ModalSubmit(modal) => modal.id,
         }
     }
+
+    /// Returns `true` if the interaction was run in a DM.
+    pub const fn is_dm(&self) -> bool {
+        match self {
+            Interaction::Ping(_) => false,
+            Interaction::ApplicationCommand(command) => command.user.is_some(),
+            Interaction::ApplicationCommandAutocomplete(command) => command.user.is_some(),
+            Interaction::MessageComponent(component) => component.user.is_some(),
+            Interaction::ModalSubmit(modal) => modal.user.is_some(),
+        }
+    }
+
+    /// Returns `true` if the interaction was run in a guild.
+    pub const fn is_guild(&self) -> bool {
+        self.guild_id().is_some()
+    }
+
+    /// Return the [`InteractionType`] of the inner interaction.
```

Comment:
We already get rustdoc's handy link in the return type, so we can keep it simple:

```suggestion
    /// Type of interaction.
```

Reply:
I half implemented this, I like "Returns" to be the first word on methods such as this